### PR TITLE
PLAT-10538: Python BDK2.0 extension app

### PR DESCRIPTION
**Pull request by symphony-elias**
### Ticket
PLAT-10538

### Description
Implemented an extension app using Python BDK2.0

### Checklist
- [x] Referenced a ticket in the PR title and in the corresponding section
- [x] Filled properly the description and dependencies, if any
- [x] Public functions properly commented

### CODE REVIEW

File: generators/bdk/python/templates/ext-app/main.py.ejs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+from python.ext_app_be import CircleOfTrustService, CircleOfTrustHandler, APP_BASE_PATH
+
+CURRENT_DIR = os.path.dirname(os.path.abspath(__file__))
+RESOURCES_FOLDER = os.path.join(CURRENT_DIR, "../resources")
+STATIC_FOLDER = os.path.join(RESOURCES_FOLDER, "static")
+
+def run_app():
+    ssl_context = ssl.create_default_context(ssl.Purpose.CLIENT_AUTH)
+    # pass the server SSL certificate and private key
+    cert_file_path = os.path.join(STATIC_FOLDER, "srv.crt")
```
Comment (symphony-youri) [inline generators/bdk/python/templates/ext-app/main.py.ejs:17]: those are autosigned certificates right? will it work out of the box everyone or will they have to authorize them or something locally?

File: test/python-bdk.spec.js [hunk 2]
```diff
@@ -26,33 +30,73 @@ describe('Python BDK', () => {
         language: 'python',
         appId: 'app-id'
       }).then((dir) => {
+        assertCommonFilesGenerated(dir);
+      })
+  })
+
+  it('Generate 2.0 python ext-app', () => {
```
Comment (symphony-youri) [inline test/python-bdk.spec.js:37]: I think for Java we are building the project part of the test, is that possible for Python too?
(the tricky thing would be the image we are using to run the tests, it has Java + Node, does it have Python...)

File: generators/bdk/python/templates/ext-app/ext_app_be.py.ejs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+        return web.Response(status=204)
+
+    async def jwt(self, request: aiohttp.web.Request):
+        body = await request.json()
+        jwt = body["jwt"]
+
+        user_id = await self.circle_of_trust_service.validate_jwt_and_retrieve_user_id(jwt)
+        response = web.json_response({"userId": user_id})
+        response.set_cookie(name="userJwt", value=jwt, path=APP_BASE_PATH, max_age=ONE_DAY_IN_SECONDS, secure=True,
```
Comment (symphony-youri) [inline generators/bdk/python/templates/ext-app/ext_app_be.py.ejs:62]: do we want to promote the use of cookies? (vs returning the JWT in the payload?)

**Review by symphony-youri — APPROVED** [review]

File: generators/bdk/python/templates/ext-app/config.yaml.ejs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+host: <%= host %>
```
Comment (symphony-thibault) [inline generators/bdk/python/templates/ext-app/config.yaml.ejs:1]: Could we reuse the same `config.taml.ejs` template across different generators?

File: generators/bdk/python/templates/ext-app/main.py.ejs (hunk 1 above)
Comment (symphony-elias) [reply generators/bdk/python/templates/ext-app/main.py.ejs:17]: Yes these are self-signed certificate extracted from https://github.com/SymphonyPlatformSolutions/generator-symphony/blob/master/generators/bdk/java/templates/ext-app/resources/keystore.p12
I could add a comment saying certs are self-signed and need to be updated

File: generators/bdk/python/templates/ext-app/ext_app_be.py.ejs (hunk 3 above)
Comment (symphony-elias) [reply generators/bdk/python/templates/ext-app/ext_app_be.py.ejs:62]: I don't know, I reused what we did for the java BDK. I can remove that part if needed.

File: generators/bdk/python/templates/ext-app/config.yaml.ejs (hunk 4 above)
Comment (symphony-elias) [reply generators/bdk/python/templates/ext-app/config.yaml.ejs:1]: Legacy projects use json config files and the java BDK2.0 ext app has a config file with a slightly different format (same config as the python config but inside a `bdk` field). There is an include option in ejs: https://ejs.co/#features but it means we will have the wrong indentation.

File: generators/bdk/python/templates/ext-app/ext_app_be.py.ejs (hunk 3 above)
Comment (symphony-elias) [reply generators/bdk/python/templates/ext-app/ext_app_be.py.ejs]: Removed

File: test/python-bdk.spec.js (hunk 2 above)
Comment (symphony-elias) [reply test/python-bdk.spec.js:37]: Opened https://perzoinc.atlassian.net/browse/PLAT-10747 for improvements

**Pull request merged**